lib/app.rb

```diff
@@ -0,0 +1,99 @@
+# rubocop:disable Style/CaseEquality
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    list.length.times do |i|
+      yield(list[i])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    list.length.times do |i|
+      yield(list[i], i)
+    end
+    self
+  end
```

Comment:
- [ ] With `my_each_with_index` is not working and has a similar issue as the `my_each` method.